src/views/ModifyPropertyView.vue

```diff
@@ -0,0 +1,137 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue';
+import { useRouter, useRoute } from 'vue-router';
+import ProjectService from '@/services/ProjectService';
+import InputText from 'primevue/inputtext';
+import Textarea from 'primevue/textarea';
+import Dropdown from 'primevue/dropdown';
+import Button from 'primevue/button';
+
+const props = defineProps<{
+  projectId: string;
+}>();
+
+const router = useRouter();
+const route = useRoute();
+const projectService = new ProjectService();
+
+const id = ref(route.query.propertyId || '');
+const title = ref('');
+const description = ref('');
+const district = ref<string>(''); // Gemarkung
+const corridor = ref<string>(''); // Flur
+const parcel = ref<string>(''); // Flurstück
+const landRegistry = ref<string>(''); // Liegenschaftsbuch
+const usageType = ref<string | null>(null); // Wirtschaftsart
+const usageOptions = [
+  { label: 'GF (Gebäude und Freifläche)', value: 'GF' },
+  { label: 'BF (Betriebsfläche)', value: 'BF' },
+  { label: 'Erholungsfläche', value: 'Erholungsfläche' },
+];
+
+onMounted(() => {
+  if (id.value) {
+    fetchPropertyDetails();
+  }
+});
+
+const fetchPropertyDetails = () => {
+  projectService
+      .getProperty(props.projectId, id.value.toString())
+      .then((property) => {
+        title.value = property.title || '';
+        description.value = property.description || '';
+        district.value = property.district || '';
+        corridor.value = property.corridor || '';
+        parcel.value = property.parcel || '';
+        landRegistry.value = property.landRegistry || '';
+        usageType.value = property.usageType ?? null;
+      })
+      .catch((err) => {
+        console.error('Fehler beim Laden der Objektdetails:', err);
+      });
+};
+
+const updateProperty = () => {
+  if (!id.value) return;
+
+  projectService
+      .updateProperty(props.projectId, id.value.toString(), {
+        title: title.value,
+        description: description.value,
+        district: district.value || '',
+        corridor: corridor.value || '',
+        parcel: parcel.value || '',
+        landRegistry: landRegistry.value || '',
+        usageType: usageType.value ?? null,
+
+
+        landRegisterEntry: '',
+        plotArea: 0,
+        effective_space: 0,
+      })
+      .then(() => {
+        router.push(`/project/${props.projectId}/objects`);
+      })
+      .catch((err) => {
+        console.error('Fehler beim Aktualisieren des Eigentums:', err);
+      });
+};
+
+
+const cancel = () => {
+  router.push(`/project/${props.projectId}/objects`);
+};
+</script>
+
+<template>
+  <div class="col-12">
+    <div class="card">
+      <h5>Bearbeite Eigentum mit ID: {{ id }}</h5>
```

Comment:
when [update url](http://localhost:5173/project/68aee5dd-fdf0-44c2-9d1e-6a424c3aed20/property/123/update) is called, `id` value is empty, even though it's supposed to be printed based on line 90. My guess is because the value of id is searched using 'params.query'. 
Is this behaviour intended?

Also, have you considered putting the `id` as props for this component? because based on my understanding, an existing property should exist in the first place for it to be modified, right?

<img width="1779" alt="Screenshot 2024-12-20 at 17 51 07" src="https://github.com/user-attachments/assets/544a0327-e9d3-4d2a-9958-ed17c79bfcfc" />